### ISSUE TITLE
Errors with "Getting Started" - setting up local environment

### ISSUE THREAD

**Issue by matosa** [freeCodeCamp/freeCodeCamp#15418]
[Contributer's Guide](https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/CONTRIBUTING.md)

#### Issue Description

When I attempt the command `npm run only-once` I get a series of errors:

```
/****/
Seeding Database
/****/
module.js:487
    throw err;
    ^
Error: Cannot find module 'babel-register'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/anicamatos/freeCodeCamp/seed/index.js:2:1)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! freecodecamp@0.1.0 only-once: `npm run create-rev && echo '/****/' && echo 'Seeding Database' && echo '/****/' && node seed && echo '/****/' && echo 'Seeding Completed' && echo '/****/'`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the freecodecamp@0.1.0 only-once script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

```
#### Browser Information

* Browser Name, Version: Chrome 58.0.3029.110 (64-bit)
* Operating System: macOS Sierra 10.12.4 (16E195)
* Mobile, Desktop, or Tablet: Laptop

**Comment by erictleung** [comment]
@matosa hi there! Looks like you might be missing some packages. Did you run the following yet?

```bash
# Install NPM dependencies
npm install

# Install Gulp globally
npm install -g gulp
```

**Comment by matosa** [comment]
I already had npm installed. I installed gulp globally and got a different error this time:
`npm ERR! missing script: only-once`

**Comment by texas2010** [comment]
Did you run mongod first then npm run ony-once?

just wondering.

**Comment by matosa** [comment]
Yes, because that's the order in which the guide specifies.

**Comment by erictleung** [comment]
@matosa did you run `npm run only-once` in the root of the freeCodeCamp repository? I believe if you run it in a different directory, it won't find the `package.json` file it needs to find the `only-once` script. So change your directory so that you have `package.json` in the same directory as you are running `npm run only-once`.

**Comment by matosa** [comment]
Yes, I ran it in the same repository as the package.json file, and I got the same error as in my original post.

**Comment by erictleung** [comment]
@matosa above, when I mentioned to run `npm install`, you replied saying you had npm installed already. This command installs the necessary modules for freeCodeCamp, which might be why the original error has the following line in it

```
Error: Cannot find module 'babel-register'
```

I don't mean to be a broken record and repeat my question, but just want to make sure I understood your statement correctly. Otherwise, I'd try to reach out to the [Contributors Help](https://gitter.im/FreeCodeCamp/Contributors) chat room.

**Comment by matosa** [comment]
I made sure to install and update npm. For some reason babel-register never installed. It wasn't in my node_modules repository when I checked manually. Unfortunately I already tried posting in the chat room, and no one has responded.

**Comment by erictleung** [comment]
@matosa I would say, delete your `node_modules` directory and then run `npm install` again to see if it downloaded this time.

**Comment by matosa** [comment]
I reinstalled my node modules and it still wasn't there, so I ran `npm install --save-dev babel-core`. Now I get this error instead:
module.js:487
    throw err;
    ^

```
Error: Cannot find module 'dotenv'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/anicamatos/freeCodeCamp/seed/index.js:3:1)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
```

**Comment by matosa** [comment]
I ended up running into a few other errors along the way, one of which were related to installing npm with homebrew. I uninstalled it by deleting my node modules, running `brew uninstall node`, and then reinstalling from the node website. I also ran into a problem where I kept getting the error: `zsh: command not found: gulp`, so I ran `npm config set prefix /usr/local` and that seemed to do the trick. Hopefully, this can help someone else who runs into the same problem.
